lib/multi.c

```diff
@@ -2370,8 +2354,10 @@ static CURLMcode singlesocket(struct Curl_multi *multi,
       }
       else {
         /* still users, but remove this handle as a user of this socket */
-        Curl_hash_delete(&entry->transfers, (char *)data,
-                         sizeof(struct Curl_easy *));
+        if(Curl_hash_delete(&entry->transfers, (char *)&data,
+                            sizeof(struct Curl_easy *))) {
+          DEBUGASSERT(NULL);
```

Comment:
Results in trap for my test case, checking my code but I haven't seen as assert on null, I need to learn it.

Reply:
The hash deletion there shouldn't fail. It indicates a problem.

Reply:
Oh ok, I overlooked the condition check 'if'